src/MICore/LaunchOptions.cs

```
@@ -63,19 +63,27 @@ public enum LaunchCompleteCommand
     /// </summary>
     public sealed class PipeLaunchOptions : LaunchOptions
     {
-        public PipeLaunchOptions(string PipePath, string PipeArguments, string PipeCommandArguments)
+        public PipeLaunchOptions(string PipePath, string PipeArguments, string PipeCommandArguments, string PipeCwd)
         {
             if (string.IsNullOrEmpty(PipePath))
                 throw new ArgumentNullException("PipePath");
 
             this.PipePath = PipePath;
             this.PipeArguments = PipeArguments;
             this.PipeCommandArguments = PipeCommandArguments;
+            if (!String.IsNullOrWhiteSpace(PipeCwd))
+            {
+                this.PipeCwd = PipeCwd;
```

Should we raise an error if this looks like a relative path or doesn't exist?